libs/client-api/src/v2/README.md

```diff
@@ -0,0 +1,929 @@
+    Note over ClientB: Apply updates<br>SV now: [A:3, B:0]
+    Note over ClientA: Disconnects
+    ClientA ->> Server: Disconnect
+    Note over Server: Remove ClientA from active participants
+```
+
+In this scenario:
+
+1. Client connects to the server and establishes a WebSocket connection
+2. Client sends a Manifest message for a document the server doesn't know about
+3. Server processes the client manifest:
+   - Checks for document existence and finds it doesn't exist
+   - Creates a new CollabGroup for this document
+   - Initializes it with an empty state vector
+   - Registers the client as an active participant
+   - Responds with a manifest containing empty state vector
+4. Client processes server manifest:
+   - Sees that server has empty state vector
+   - Encodes its entire document state as an update
+   - Sends this complete update to the server (no need to request missing updates)
+5. Server processes the client update:
+   - Applies the complete update to the empty document
+   - Updates its state vector to reflect the client's document state [A:3]
+6. When other clients connect, the server:
+   - Recognizes that the document now exists
+   - Compares the new client's state vector with its own
+   - Sends a manifest with current state vector
+   - Processes the client's response and provides missing updates
+
+The server maintains a list of active participants (subscribers) for each document, which allows it to:
+
+- Send updates to all connected clients when changes occur
+- Clean up resources when clients disconnect
+- Handle reconnections efficiently
+
+Key differences between the two scenarios:
+
+- In Case 1, the server recognizes an existing document and provides its state vector for comparison
+- In Case 2, the server creates a new document and initializes it with the client's document state
+- In both cases, the server compares state vectors to determine what updates need to be exchanged
+- In both cases, the server maintains a registry of active clients for broadcasting updates
+
+When a client makes changes in normal operation:
+
+1. Changes are made locally first and immediately applied to the local document
+2. The client's state vector is updated to reflect the new operation
+3. The update is sent to the server with a message ID (Rid) and the current state vector
+4. The server processes the update through its group-based architecture:
+    - `handle_client_collab_message` validates the message and user
+    - `forward_message_to_group` routes it to the appropriate CollabGroup
+    - `handle_protocol_message` processes the message based on its type
+    - The update is applied to the server's document
+5. The server broadcasts the update to all other connected clients of that document
+6. Each receiving client:
+    - Applies the update to their local document
+    - Updates their state vector and last_message_id
+
+### Offline Editing & Reconnection
```

Comment:
Tbh. there's no need to describe 3 flows with diagrams. In practice it's a single flow of messages, regardless of where the document is present or what was the connection state before. Because of that this protocol description is bigger than it actually needs.